### DISCUSSION TITLE
About fault to selection ratio

### DISCUSSION THREAD

**Discussion by FasihMunirMalik** [christianbirchler-org/sdc-testing-competition#24] [Q&A] [answered]
The fault-to-selection ratio being used as an evaluation metric returns: 
`return nr_detected_faults / len(test_suite) `
where as it should be `nr_detected_faults / len(selection)`

so incase of a test suite of 192 cases where I select 74, 51 out of these 74 fail so fault to selection ratio should be 51/74, currently It is 51/192

**Comment by ChristianBirchler** [comment, marked as answer]
Thanks @FasihMunirMalik ! It'll be fixed (#27 ).

**Reply by FasihMunirMalik** [reply]
@ChristianBirchler 
There is a missing package in Docker image, I was unable to assign this as an issue to you
**Description**:  
A recent update introduced the use of `scikit-learn` in the evaluation.py file, specifically functions like `pairwise_distances` from `sklearn.metrics.pairwise`. However, `scikit-learn` is not currently installed in the Evaluator Docker image, which leads to an import error when running the container.

**Steps to Reproduce**:
1. Build latest Docker image.
2. Run the container.
3. Observe the import error for `scikit-learn`.

**Current Behavior**:  
Running the Docker container results in an error indicating that `scikit-learn` is missing.

**Expected Behavior**:  
The Docker image should have `scikit-learn` installed so that functions like `pairwise_distances` work without errors.

**Suggested Fix**:  
Add `scikit-learn` to the Evaluator Dockerfile. For example, you could include:
   ```dockerfile
   RUN pip install scikit-learn
   ```

This would ensure that `scikit-learn` is available in the container, allowing for successful execution of scripts that depend on it.

Thank you for addressing this issue!
